### ISSUE TITLE
Context(this) is undefined when used in dynamic ES module import.

### ISSUE THREAD

**Issue by klemenoslaj** [Microsoft/TypeScript#19199]
Tested on:
* Version 2.5.3
* Version 2.6.0-dev.20171015

**Code**
**TypeScript**
```ts
class Test { 
    path = './some/file';

    load() {
      import(this.path)
          .then(() => console.log('loaded'))
          .catch(() => console.log('not loaded'));
    }
}
```
Compiled **JavaScript**
```js
define(["require", "exports"], function (require, exports) {
    "use strict";
    var Test = /** @class */ (function () {
        function Test() {
            this.path = './some/file';
        }
        Test.prototype.load = function () {
            new Promise(function (resolve_1, reject_1) { require([this.path], resolve_1, reject_1); }).then(function () { return console.log('loaded'); })
                .catch(function () { return console.log('not loaded'); });
        };
        return Test;
    }());
});
```

**Expected behavior:**
When using a property `path` of context `this` as a parameter to `import`, `this` should be an instance of `Test` class.

**Actual behavior:**
`this` is wrapped in a `function`, thus referring to `undefined` in strict mode. Should be using cached `_this` as in other similar cases.

**Comment by weswigham** [comment]
@mhegazy Is this not fixed in #19057, duplicating #17564?

**Comment by klemenoslaj** [comment]
@weswigham, indeed it looks like a duplicate, sorry.
